src/collection.ts

```diff
@@ -975,12 +998,111 @@ export class Collection<TSchema extends Document = Document> {
    */
   async count(filter: Filter<TSchema> = {}, options: CountOptions = {}): Promise<number> {
     return executeOperation(
-      this.s.db.s.client,
-      new CountOperation(
-        MongoDBNamespace.fromString(this.namespace),
-        filter,
-        resolveOptions(this, options)
-      )
+      this.client,
+      new CountOperation(this.fullNamespace, filter, resolveOptions(this, options))
+    );
+  }
+
+  /**
+   * @internal
```

Comment:
Should these be @internal? Once released we're going to have to use them from mongosh, right?

Reply:
That's fine, I guess. I forked this branch so I can work with it locally in the meantime anyway.